jdk_11_maven/em/embedded/rest/pay-publicapi/src/main/java/em/embedded/uk/gov/pay/api/app/EmbeddedEvoMasterController.java

```diff
@@ -0,0 +1,152 @@
+import org.evomaster.client.java.controller.api.dto.AuthenticationDto;
+import org.evomaster.client.java.controller.api.dto.SutInfoDto;
+import org.evomaster.client.java.controller.problem.ProblemInfo;
+import org.evomaster.client.java.controller.problem.RestProblem;
+import org.evomaster.client.java.sql.DbSpecification;
+import org.testcontainers.containers.GenericContainer;
+import uk.gov.pay.api.app.PublicApi;
+
+import java.io.File;
+import java.util.List;
+
+public class EmbeddedEvoMasterController extends EmbeddedSutController {
+
+    private static final int REDIS_PORT = 6379;
+
+    private static final String REDIS_VERSION = "7.2.3";
+
+    private static final GenericContainer redisContainer = new GenericContainer("redis:" + REDIS_VERSION)
+            .withExposedPorts(REDIS_PORT);
+
+    public static void main(String[] args) {
+        int port = 40100;
+        if (args.length > 0) {
+            port = Integer.parseInt(args[0]);
+        }
+
+        EmbeddedEvoMasterController controller = new EmbeddedEvoMasterController(port);
+        InstrumentedSutStarter starter = new InstrumentedSutStarter(controller);
+
+        starter.start();
+    }
+
+    private PublicApi application;
+
+    public EmbeddedEvoMasterController() {
+        this(40100);
+    }
+
+    public EmbeddedEvoMasterController(int port) {
+        setControllerPort(port);
+    }
+
+    @Override
+    public boolean isSutRunning() {
+        if (application == null) {
+            return false;
+        }
+
+        return application.getJettyServer().isRunning();
+    }
+
+    @Override
+    public String getPackagePrefixesToCover() {
+        return "uk.gov.pay.api.app.";
+    }
+
+    @Override
+    public List<AuthenticationDto> getInfoForAuthentication() {
+        return null;
+    }
+
+    @Override
+    public ProblemInfo getProblemInfo() {
+        return new RestProblem(
+                "http://localhost:" + application.getJettyPort() + "/api/swagger.json",
+                null
+        );
+    }
+
+    @Override
+    public SutInfoDto.OutputFormat getPreferredOutputFormat() {
+        return SutInfoDto.OutputFormat.JAVA_JUNIT_5;
+    }
+
+    @Override
+    public String startSut() {
+
+        redisContainer.start();
+
+//        REDIS_URL
+        application = new PublicApi();
+
+        //Dirty hack for DW...
+        System.setProperty("dw.server.applicationConnectors[0].port", "0");
+        System.setProperty("dw.server.adminConnectors[0].port", "0");
+        System.setProperty("dw.redis.endpoint", "localhost:" + REDIS_PORT);
+
+        /*
+        Note: When running using IntelliJ, make sure the working directory is set to the
+        driver module.
+         */
+        try {
+            application.run("server", "src/main/resources/em_config.yaml");
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+
+        try {
+            Thread.sleep(3_000);
```

Comment:
you ll need to check if this value is valid for this API...

Reply:
actually fine... but see if in new version of Dropwizar `run` is blocking until API is initialized